src/System.Net.Http/src/uap/System/Net/HttpClientHandler.cs

```diff
@@ -684,6 +714,63 @@ private static bool InitRTNoCacheSupported()
                 "NoCache");
         }
 
+        private void RTServerCertificateCallback(RTHttpBaseProtocolFilter sender, RTHttpServerCustomValidationRequestedEventArgs args)
+        {
+            // Convert WinRT certificate to .NET certificate.
+            X509Certificate2 serverCert = ConvertCertificate(args.ServerCertificate);
+
+            // Create .NET X509Chain from the WinRT information. We need to rebuild
+            // the chain since WinRT only gives us an array of intermediate certificates.
+            var serverChain = new X509Chain();
+            foreach (RTCertificate cert in args.ServerIntermediateCertificates)
+            {
+                serverChain.ChainPolicy.ExtraStore.Add(ConvertCertificate(cert));
+            }
+            serverChain.Build(serverCert);
+
+            // Determine chain and name errors from WinRT information.
+            SslPolicyErrors sslPolicyErrors = SslPolicyErrors.None;
+            if (serverChain.ChainStatus.Length > 0)
+            {
+                sslPolicyErrors |= SslPolicyErrors.RemoteCertificateChainErrors;
+            }
+            foreach (RTChainValidationResult result in args.ServerCertificateErrors)
```

Comment:
I'd like @bartonjs to review the "chain building" code here.  I didn't use the code in SslStream and WinHttpHandler because that code also uses Chain.Build to build the chain. The only reason it then uses the low-level SSPI calls is to find out if the name mismatch occurs.  The current .NET X509Chain.Build doesn't take any DNS name to compare against. On the other hand, this information is already available in the WinRT ChainValidationResult. So, that means, I don't need to call the SSPI stuff since I already have the potential name-mismatch determination.